### ISSUE TITLE
Config flow could not be loaded: undefined

### ISSUE THREAD

**Issue by navarro152** [jseidl/hass-magic_areas#215]
Hello, 

I am getting the following error when I press "Configure" in certain areas "Config flow could not be loaded: undefined". It shows briefly the loading window and then I get the error one. Other areas are working besides two that I configure previously (the other ones working I haven't configured them yet.
I already tried restarting HA, restating all the system and even the full server where I have my HA installation. As an addition, I delete all the areas and the custom integration, but I still got the same error after I installed it again.

Any clue how can I solve it? I am adding the logs that I get after I click the button configure in one of the areas that I want to configure that is giving me the error.

```
2022-06-12 20:33:26 ERROR (MainThread) [aiohttp.server] Error handling request
Traceback (most recent call last):
  File "/usr/local/lib/python3.9/site-packages/aiohttp/web_protocol.py", line 435, in _handle_request
    resp = await request_handler(request)
  File "/usr/local/lib/python3.9/site-packages/aiohttp/web_app.py", line 504, in _handle
    resp = await handler(request)
  File "/usr/local/lib/python3.9/site-packages/aiohttp/web_middlewares.py", line 117, in impl
    return await handler(request)
  File "/usr/src/homeassistant/homeassistant/components/http/security_filter.py", line 60, in security_filter_middleware
    return await handler(request)
  File "/usr/src/homeassistant/homeassistant/components/http/forwarded.py", line 100, in forwarded_middleware
    return await handler(request)
  File "/usr/src/homeassistant/homeassistant/components/http/request_context.py", line 28, in request_context_middleware
    return await handler(request)
  File "/usr/src/homeassistant/homeassistant/components/http/ban.py", line 79, in ban_middleware
    return await handler(request)
  File "/usr/src/homeassistant/homeassistant/components/http/auth.py", line 220, in auth_middleware
    return await handler(request)
  File "/usr/src/homeassistant/homeassistant/components/http/view.py", line 137, in handle
    result = await result
  File "/usr/src/homeassistant/homeassistant/components/config/config_entries.py", line 243, in post
    return await super().post(request)
  File "/usr/src/homeassistant/homeassistant/components/http/data_validator.py", line 62, in wrapper
    result = await method(view, request, *args, **kwargs)
  File "/usr/src/homeassistant/homeassistant/helpers/data_entry_flow.py", line 70, in post
    result = await self._flow_mgr.async_init(
  File "/usr/src/homeassistant/homeassistant/data_entry_flow.py", line 205, in async_init
    flow, result = await task
  File "/usr/src/homeassistant/homeassistant/data_entry_flow.py", line 232, in _async_init
    result = await self._async_handle_step(flow, flow.init_step, data, init_done)
  File "/usr/src/homeassistant/homeassistant/data_entry_flow.py", line 335, in _async_handle_step
    result: FlowResult = await getattr(flow, method)(user_input)
  File "/config/custom_components/magic_areas/config_flow.py", line 130, in async_step_init
    CONF_MAIN_LIGHTS: cv.multi_select(sorted(all_lights)),
TypeError: '<' not supported between instances of 'list' and 'str'

```

**Comment by hyperfive** [comment]
I am having the same issue.  This occured after I changed the entity id on items that were part of magic area, then when it would no longer allow me to open the configuration page for that magic area I deleted it and restarted HA.

**Comment by hyperfive** [comment]
@navarro152 - I found a workaround for the areas that are coming up with this message.  It's not ideal but it works.  

1. Rename old area 
2. create a new area with the original name 
3. move all devices/entities into that new area.
4. restart HA

The interesting thing is once I did this the old area started working again.

**Comment by jseidl** [comment]
This is unfortunately a known issue (#156) that I have yet to solve with config_flow. If an entity is renamed or removed **while** it's configured in MA, it breaks config_flow.

Please watch that thread to know once this is fixed! Sorry about the inconvenience

**Comment by jseidl** [comment]
@navarro152 does the steps from @hyperfive works for you?

**Comment by navarro152** [comment]
@hyperfive I tried this workaround. It seemed fine, but I see that there are some issues.
1) It seems to work fine, but I cannot edit the part of "main lights", it shows as blanc, and in other areas only shows one light. 
2) After removing the old area, it gave me an error and it does not start anymore the config options. Here I tried with another area, and it seems to work fine until I remove the old area as well.

*For each try I restarted HA

Error logs
`Entity ID is not a string: ['light.kitchen_cabinet_light', 'light.kitchen_fan_light_1', 'light.kitchen_fan_light_2', 'light.kitchen_fan_light_3', 'light.kitchen_sink_light']
4:00:27 PM – (WARNING) Magic Areas (custom integration) - message first occurred at 4:00:27 PM and shows up 14 times
Detected integration that uses deprecated `async_get_registry` to access entity registry, use async_get instead. Please report issue to the custom component author for shelly using this method at custom_components/shelly/__init__.py, line 356: await self.hass.helpers.entity_registry.async_get_registry()
4:00:22 PM – (WARNING) helpers/frame.py - message first occurred at 3:58:50 PM and shows up 5 times
Error setting up entry Living Room for magic_areas
3:58:56 PM – (ERROR) Magic Areas (custom integration) - message first occurred at 3:58:56 PM and shows up 3 times`

@jseidl I was hoping, but I got another error that does not let me open the configs again.

**Comment by bmcclure** [comment]
I'm running into this but only in one area of my house currently. 

While troubleshooting, I deleted the area in MA, restarted HA, and then I was able to configure the broken area in MA again. But then an hour later I went back into configure it again because the area wasn't working right, and I am back to receiving this config flow error again even though nothing changed in my HA configuration between that time.

Is there a better workaround for this than actually creating a whole new area in HA? Why wouldn't deleting the area in Magic Areas and letting it be re-created automatically be sufficient? Thanks!

**Comment by jseidl** [comment]
This is a known bug (if it's the same thing) when you have a group in an area. What happens is that HA replaces the entity_id attribute with a list with all entities in the group and that is not what MA is expecting so it browns out. 

I've been pretty busy lately (just had a kid) so I'm hoping this weekend I can make some time to fix this issue and the new one that came after breaking change on update

**Comment by jseidl** [comment]
Fixed on 3.0. If the issue persists, please reopen!

**Comment by navarro152** [comment]
Congratulations for your kid and thanks for still put the time on this! I will try to configure them again :)
